File: lib-jvm/src/main/java/com/n8cats/share/data/BigAction.java
package com.n8cats.share.data;
import com.n8cats.share.Logic;

import org.jetbrains.annotations.Nullable;

public class BigAction implements Logic.InStateAction {//todo redundant because Json serialization
	@Nullable public NewCarAction n;
	@Nullable public PlayerAction p;
	public void act(State state, Logic.GetCarById getCar) {
		if(n != null) n.act(state, getCar);
		if(p != null) p.act(state, getCar);
	}
}

File: server-common/src/main/kotlin/com/riseofcat/share/data/BigAction.kt
package com.riseofcat.share.data

import kotlinx.serialization.*

@Serializable class BigAction:InStateAction {
  @Optional var n:NewCarAction? = null
  @Optional var p:PlayerAction? = null

  override fun act(state:State,getCar:GetCarById) {
    n?.act(state,getCar)
    p?.act(state,getCar)
  }
}
